### ISSUE TITLE
Fatal error while calling CouchbaseBucket->get

### ISSUE THREAD

**Issue by chetan-prime** [brett19/php-couchbase#18]
I have php code using php-couchbase working well on my webUI. However I noticed an issue since upgrading to php-couchbase 2.1.0 release. It seems to crash when I try to get a deleted / non-existent document.
Code that gives fatal error -

$record=$this->bucket->get($docid);
PHP Fatal error:  Cannot use object of type stdClass as array in [CouchbaseNative]/CouchbaseBucket.class.php on line 284

Expected result -
CouchbaseException thrown indicating the record does not exist. Does the fact that the document in question was deleted make a difference? In any case it calls for some further sanity checks in the CouchbaseBucket.class.php to avoid fatal errors in unexpected conditions.

**Comment by SergeyKozlov** [comment]
Hello.
I do so:

```
            try {
                $res = $bucket->get($docid);
                return $res->value;
            } catch (Exception $e) {
                echo "Not found. " . $e->getMessage();
                return false;
            }
```

**Comment by chetan-prime** [comment]
Hi @SergeyKozlov ,
I use your method by catching \CouchbaseException , but in this case php gives a Fatal Error and terminates the script, so there is no way we can catch it.
Usually the exception handling works fine , but in this case the key was previously deleted so I don't know if that triggers an error in the couchbase php code.

**Comment by SergeyKozlov** [comment]
Yes you are right.
I look for a case where I caught this error.

**Comment by SergeyKozlov** [comment]
All I understood.
When I use the code:

```
            try {
                $res = $bucket->get("fake");
                return $res->value;
            } catch (Exception $e) {
                echo "Not found. " . $e->getMessage();
                return false;
            }
```

I get an answer:
**Not found. The key does not exist on the server**
The program continues to run.
Try to check the settings of php.
And try to use it here

```
                //error_reporting(0); // Turn off error reporting
                error_reporting(E_ALL); // Report all errors
```

Check your version Couchbase SDK.

**Comment by chetan-prime** [comment]
This error comes usually for me  in non-existent keys and I can handle the exception. However the error I get is only with some particular keys which existed before but are now deleted

**Comment by SergeyKozlov** [comment]
Maybe it is to explain the situation:
[https://forums.couchbase.com/t/get-why-is-change-the-data-type/6723]

**Comment by chetan-prime** [comment]
Please note the fatal error occurs within php-couchbase module, not in my code.
The error occurs within the bucket->get  call itself and the fuction never returns. I am inclined to think its a bug in stub/CouchbaseBucket.class.php
Also i never edited the key in webUI. The key is deleted using the bucket->remove method and doesn't even show up in the list of keys in the couchbase admin webUI

**Comment by SergeyKozlov** [comment]
Name key is UTF-8 ?

**Comment by chetan-prime** [comment]
Thanks for the prompt follow up.
This is the key -->  user_uuid-internal-55f84f904105b
It's alphanumeric only no special chars. Have a lot of other keys in the same series and they all work fine except the ones that got deleted now crash php-couchbase

**Comment by chetan-prime** [comment]
Is anyone able to replicate my issue? That when you try to access some(maybe all) deleted keys then it crashes php-couchbase with a Fatal Error in the inbuilt file CouchbaseBucket.class.php

**Comment by chetan-prime** [comment]
I see @brett19 had pushed a commit 916d4b7f6c31f03662b43e9e8cd08500087aae1a back in Nov. to master, I'm trying the patch from master and will confirm if that fixed the issue. Was just luck that I saw this commit right at the top of the repo browser.

**Comment by brett19** [comment]
Hey @chetan-prime,
That commit is actually related to our N1QL querying capabilities (`query()`) in the SDK rather than the normal Key-Value APIs (`get`/`upsert`/`remove`).  I am unfortunately having trouble reproducing this issue.  Could you perhaps give me a short example script which can reproduce this issue?
Cheers, Brett

**Comment by chetan-prime** [comment]
Hi @brett19 ,
I can confirm that commit 916d4b7f6c31f03662b43e9e8cd08500087aae1a solves my problem. I just updated 2 files phpstubstr.h & CouchbaseBucket.class.php into the 2.10 sources and compiled once again. The code in question is below -

$record=$this->bucket->get($docid);

where $docid="internal-55f84f904105b" , This document previously existed and was deleted a few days back. Before I used the new stub files it was crashing exactly at stub/CouchbaseBucket.class.php line 284. I don't know how a call to ->get ends up in ->_view . Looks like you are internally building a query even for a ->get statement?
Anyway the check you put in for json mode seems to have solved the problem. Looks like in certain cases like some deleted records you are getting an object instead of an array and that was hard crashing php with an uncatchable fatal error.
Thanks, that commit worked for me
